flink/json-jobs.yml

```diff
@@ -0,0 +1,119 @@
+type: json
+data:
+  url: /v1/jobs/metrics?get=numberOfFailedCheckpoints,lastCheckpointSize,lastCheckpointExternalPath,totalNumberOfCheckpoints,lastCheckpointRestoreTimestamp,lastCheckpointAlignmentBuffered,restartingTime,uptime,numberOfInProgressCheckpoints,downtime,numberOfCompletedCheckpoints,fullRestarts,lastCheckpointDuration
+  server: ${FLINK_HOSTPORT}
+
+observation:
+  - name: jobs_uptime
+    metricNamespace: flink
+    path: $.[?(@.id=uptime)]
+    metric:
+      - name: jobs.uptime
+        source: sum
+        type: gauge
+        label: uptime
+        description: The time that the job has been running without interruption
+
+  - name: jobs_downtime
+    metricNamespace: flink
+    path: $.[?(@.id=downtime)]
+    metric:
+      - name: jobs.downtime
+        source: sum
+        type: gauge
+        label: downtime
+        description: "For jobs currently in a failing/recovering situation, the time elapsed during this outage. 0 for running jobs and -1 for completed jobs"
+
+  - name: jobs_restartingtime
+    metricNamespace: flink
+    path: $.[?(@.id=restartingTime)]
+    metric:
+      - name: jobs.restartingtime
+        source: sum
+        type: gauge
```

Comment:
Gauge or counter?

Reply:
it is counter, but when job finish it will return -1. is there any problem in this case. 
same to uptime, downtime.

Reply:
For how long will it return -1?

When counter runs into -1 it reports 0 and sets the new "base" to -1. So once it starts growing again, our counter will return value bigger than in reality by "1", but that is minor issue. Bigger issue is will it even run into -1, because if the number starts incrementing before agent fetches -1, numbers will be even more incorrect.

Reply:
I think it can be solved by converting -1 to 0, can we do that by a function. @bsmid

Reply:
There is `IfThenElse` function. But I think it will not help us, problem we have is "if the number starts incrementing before agent fetches -1".

Reply:
It return -1 when job completed (for uptime & downtime) so it always return -1 after job completed.
For restartingTime, it doesn't return -1.
So i will use IfThenElse for uptime, downtime
Is my syntax correct to use IfThenElse tor change -1 to 0 @bsmid 
func:IfThenElse(sum, =, -1, 0L, metric:sum)